fixed-hash/src/tests.rs

```diff
@@ -0,0 +1,327 @@
+                0x89, 0xAB, 0xCD, 0xEF,
+                0x09, 0x08, 0x07, 0x06,
+                0x05, 0x04, 0x03, 0x02
+            ]).to_low_u64_be(),
+            0x0908070605040302
+        );
+        assert_eq!(
+            H128::from([
+                0x01, 0x23, 0x45, 0x67,
+                0x89, 0xAB, 0xCD, 0xEF,
+                0x09, 0x08, 0x07, 0x06,
+                0x05, 0x04, 0x03, 0x02
+            ]).to_low_u64_le(),
+            0x0203040506070809
+        )
+    }
+}
+
+#[cfg(feature = "byteorder-support")]
+mod from_low_u64 {
+    use super::*;
+
+    #[test]
+    fn smaller_size() {
+        assert_eq!(
+            H32::from_low_u64_be(0x0123_4567_89AB_CDEF),
+            H32::from([0x01, 0x23, 0x45, 0x67])
+        );
+        assert_eq!(
+            H32::from_low_u64_le(0x0123_4567_89AB_CDEF),
+            H32::from([0xEF, 0xCD, 0xAB, 0x89])
+        );
+    }
+
+    #[test]
+    fn equal_size() {
+        assert_eq!(
+            H64::from_low_u64_be(0x0123_4567_89AB_CDEF),
+            H64::from([
+                0x01, 0x23, 0x45, 0x67,
+                0x89, 0xAB, 0xCD, 0xEF
+            ])
+        );
+        assert_eq!(
+            H64::from_low_u64_le(0x0123_4567_89AB_CDEF),
+            H64::from([
+                0xEF, 0xCD, 0xAB, 0x89,
+                0x67, 0x45, 0x23, 0x01
+            ])
+        )
+    }
+
+    #[test]
+    fn larger_size() {
+        assert_eq!(
+            H128::from_low_u64_be(0x0123_4567_89AB_CDEF),
+            H128::from([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0x01, 0x23, 0x45, 0x67, 0x89, 0xAB, 0xCD, 0xEF
+            ])
+        );
+        assert_eq!(
+            H128::from_low_u64_le(0x0123_4567_89AB_CDEF),
+            H128::from([
+                0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                0xEF, 0xCD, 0xAB, 0x89, 0x67, 0x45, 0x23, 0x01
+            ])
+        )
+    }
+}
+
+#[cfg(feature = "rand-support")]
+mod rand {
+    use super::*;
+    use rand::{XorShiftRng, SeedableRng};
+
+    #[test]
+    fn random() {
+        let default_seed = <XorShiftRng as SeedableRng>::Seed::default();
+        let mut rng = XorShiftRng::from_seed(default_seed);
+        assert_eq!(
+            H32::random_using(&mut rng),
+            H32::from([0x43, 0xCA, 0x64, 0xED])
+        );
+    }
+
+    #[test]
+    fn randomize() {
+        let default_seed = <XorShiftRng as SeedableRng>::Seed::default();
+        let mut rng = XorShiftRng::from_seed(default_seed);
+        assert_eq!(
+            {
+                let mut ret = H32::zero();
+                ret.randomize_using(&mut rng);
+                ret
+            },
+            H32::from([0x43, 0xCA, 0x64, 0xED])
+        )
+    }
+}
+
+#[cfg(feature = "rustc-hex-support")]
+mod from_str {
+    use super::*;
+
+	#[test]
+	fn valid() {
+		use core::str::FromStr;
+
+        assert_eq!(
+            H64::from_str("0123456789ABCDEF").unwrap(),
```

Comment:
What will it do if given less than 8 bytes? In any case, I think a test for that case won't hurt.